processor/filterprocessor/README.md

```diff
@@ -310,15 +310,43 @@ See the table below for details on each context and the fields it exposes.
 The OTTL allows the use of `and`, `or`, and `()` in conditions.
 See [OTTL Boolean Expressions](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/README.md#boolean-expressions) for more details.
 
-The filter processor has access to all the [factory functions of the OTTL](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#ottl-functions) 
-
 For conditions that apply to the same signal, such as spans and span events, if the "higher" level telemetry matches a condition and is dropped, the "lower" level condition will not be checked.
 This means that if a span is dropped but a span event condition was defined, the span event condition will not be checked.
 The same relationship applies to metrics and datapoints.
 
 If all span events for a span are dropped, the span will be left intact.
 If all datapoints for a metric are dropped, the metric will also be dropped.
 
+### OTTL Functions
+
+The filter processor has access to all the [factory functions of the OTTL](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#ottl-functions)
+
+In addition, the processor defines a few of its own functions:
+
+**Metrics only functions**
+- [HasAttrOnDatapoint](#HasAttrOnDatapoint)
+- [HasAttributeOnDatapointWithValue](#HasAttributeOnDatapointWithValue)
+
+#### HasAttrOnDatapoint
+
+`HasAttrOnDatapoint(key)`
+
+Returns `true` if the given key appears in the attribute map of any datapoint on a metric.
+
+Examples:
+
+- `HasAttrOnDatapoint("http.target")`
+
+#### HasAttributeOnDatapointWithValue
+
+`HasAttributeOnDatapointWithValue(key, value)`
```

Comment:
Only concern is about the naming here, can we consider something like `HasAttrOnDatapoint` where if not provided the second argument acts like current HasAttrOnDatapoint otherwise like HasAttributeOnDatapointWithValue?

Reply:
Ya the names were a concern I had as well. I went with the most literally interpretation of the existing functionality. 

OTTL doesn't have optional parameters, so to merge them into one we'd have to use a nil value in the second parameter to imply only the presence should be checked.

Reply:
This is just an idea, not sure if we want this model or not...

Reply:
Ok, some things that I have in mind:

1. The use of abbreviation `Attr` for `Attributes` is inconsistent, and I think this must be fixed one way or the other. 
2. Alternative names `HasAttrKeyOnDatapoint` and `HasAttrOnDatapoint`. 

Do we have something like this functionality (not for a sub-domain) in the core package? How do we name them there?

Reply:
Nothing similar on in ottlfuncs. All the functions there are generic enough that attributes/attr hasn't been in a name. 

We do use `attributes` as the path name for the accessor.

Reply:
I don't think that should stop us from using Attr here tho as the names are sooo long.

Reply:
So even if you don't go with my alternative proposal, you should then fix Has**Attr**OnDatapoint otherwise like Has**Attribute**OnDatapointWithValue

Reply:
The README actually had the name wrong to begin with so glad we changed it.  Updated to `HasAttrKeyOnDatapoint` and `HasAttrOnDatapoint`.